coderunners/util.py

```diff
@@ -15,6 +15,7 @@ def save_code(save_dir: Path, code: dict[str, str]) -> list[Path]:
     for filename, content in code.items():
         path = save_dir / filename
         saved_paths.append(path)
+        path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
Let's add a TODO comment here. The interface for code should be `dict[str, str | dict]`. And we should save the code as a file tree recursively.